### PR TITLE
fixed newlines

### LINKED ISSUE #2325: fix missing newlines
there are newlines missing in stable

see https://github.com/Open-MSS/MSS/pull/2323

### PR DESCRIPTION
**Pull request by ReimarBauer**
**Purpose of PR?**:

Fixes #2325

### CODE REVIEW

File: localbuild/README.rst [hunk 1]
```diff
@@ -2,4 +2,4 @@ localbuild is mainly a copy of the build instructions from
 https://github.com/conda-forge/mss-feedstock/tree/master/recipe
 
 You may want to copy this directory to a different directory on your system
-and alter it for your needs.
\ No newline at end of file
+and alter it for your needs.
```
Comment (ReimarBauer) [inline localbuild/README.rst:5]: there is also a newline added

rst maybe rendered differently here?

Comment (ReimarBauer) [reply localbuild/README.rst:5]: aha looks different in vim

**Comment by ReimarBauer** [comment]
Not thought before on that. Do we prefere `\n` in a newline or at the end of the last line?

**Comment by matrss** [comment]
> Not thought before on that. Do we prefere `\n` in a newline or at the end of the last line?

Every line should end in `\n`. This is the usual convention on Linux systems and a necessity to have commands like `wc -l` and `cat` work properly (where "properly" means not outputting surprising and potentially confusing stuff). GitHub shows its warning sign when this `\n` is missing.

Having another `\n` on a line by itself would mean two newlines at the end of the file. That is usually not wanted, I think (VSCodium would remove additional newlines like this).

**Comment by ReimarBauer** [comment]
Hmm, that's why I used vim. Redoing in pycharm.

File: localbuild/README.rst (hunk 1 above)
Comment (ReimarBauer) [reply localbuild/README.rst:5]: in pycharm there is a `\n` defined. 
image
it looks like it does somehow render rst here or at least did not show the icon for missing of a newline

**Comment by matrss** [comment]
Do we want to introduce a linter for these kinds of things? We could setup a https://editorconfig.org/ configuration for the repository and use https://github.com/editorconfig-checker/editorconfig-checker to enforce it.

**Comment by ReimarBauer** [comment]
> Do we want to introduce a linter for these kinds of things? We could setup a https://editorconfig.org/ configuration for the repository and use https://github.com/editorconfig-checker/editorconfig-checker to enforce it.

Yes, that sounds nice.  We will see if there are some windows linebreaks missed to change from the past.

**Review by matrss — APPROVED** [review]
LGTM for the scope of this PR. I filed an issue to introduce a lint workflow.

**Comment by matrss** [comment]
> Yes, that sounds nice. We will see if there are some windows linebreaks missed to change from the past.

I quickly tried it out, there are some files remaining that use windows-style linebreaks, but not many:
```
$ editorconfig-checker 
docs/make.bat:
        Wrong line endings or no final newline
        Not all lines have the correct end of line character
mslib/mswms/xml_templates/get_capabilities130.pt:
        Wrong line endings or no final newline
        Not all lines have the correct end of line character
mslib/mswms/xml_templates/service_exception130.pt:
        Wrong line endings or no final newline
        Not all lines have the correct end of line character
tests/_test_utils/test_version.py:
        Wrong line endings or no final newline
        Not all lines have the correct end of line character
tutorials/textfiles/tutorial_waypoints.txt:
        Wrong line endings or no final newline
        Not all lines have the correct end of line character

10 errors found
```

This is with a very simple .editorconfig configuration:
```
$ cat .editorconfig 
root = true

# Unix-style newlines with a newline ending every file
[*]
end_of_line = lf
insert_final_newline = true
```

**Comment by ReimarBauer** [comment]
fixing this with the new checker, is also a good introduction for it.

**Pull request merged**